src/switch_ivr_bridge.c

```diff
@@ -52,24 +52,34 @@ static void text_bridge_thread(switch_core_session_t *session, void *obj)
 	struct vid_helper *vh = obj;
 	switch_status_t status;
 	switch_frame_t *read_frame = 0;
-	switch_channel_t *channel = switch_core_session_get_channel(vh->session_a);
-	switch_channel_t *b_channel = switch_core_session_get_channel(vh->session_b);
+	switch_channel_t *channel;
+	switch_channel_t *b_channel;
 	switch_buffer_t *text_buffer = NULL;
 	switch_size_t text_framesize = 1024, inuse = 0;
 	unsigned char *text_framedata = NULL;
 	switch_frame_t frame = { 0 };
+	switch_core_session_t * session_a, *session_b;
+
+	if ((session_a = switch_core_session_locate(vh->session_a_uuid))) {
```

Comment:
what is going to happen if you can't locate the session?

Reply:
technically, the channel variable would be null and it asserts in switch_channel_up_nosig(which I don't think it's the best approach to assert here honeslty) but it won't enter the while loop.